eth/handler.go

```diff
@@ -452,9 +455,18 @@ func (pm *ProtocolManager) BroadcastBlock(block *types.Block, propagate bool) {
 
 	// If propagation is requested, send to a subset of the peer
 	if propagate {
+		// Calculate the TD of the block (it's not imported yet, so block.Td is not valid)
+		var td *big.Int
+		if parent := pm.chainman.GetBlock(block.ParentHash()); parent != nil {
+			td = new(big.Int).Add(parent.Td, block.Difficulty())
+		} else {
+			glog.V(logger.Error).Infof("propagating dangling block #%d [%x]", block.NumberU64(), hash[:4])
```

Comment:
That't the reason for the else branch and erroring out :P. But it should not happen because before propagating we check that the parent exists (https://github.com/ethereum/go-ethereum/blob/4f95e2f9ec3e416773e6a87b3c76667d97894f36/eth/fetcher/fetcher.go#L420). This is just a sanity check for future modifications if we actually call the broadcast from multiple places.